src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -1,30 +1,37 @@
 package org.broadinstitute.dsde.workbench.leonardo.model
 
+import java.net.URL
 import java.time.Instant
 import java.util.UUID
 
 import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.StatusCodes
 import com.typesafe.config.ConfigFactory
 import net.ceedubs.ficus.Ficus._
 import org.broadinstitute.dsde.workbench.leonardo.config.{DataprocConfig, ProxyConfig}
 import org.broadinstitute.dsde.workbench.leonardo.model.ClusterStatus.ClusterStatus
-import org.broadinstitute.dsde.workbench.leonardo.model.ModelTypes.{GoogleBucket, GoogleBucketUri, GoogleProject, GoogleServiceAccount}
-import spray.json.{DefaultJsonProtocol, DeserializationException, JsString, JsValue, JsonFormat}
+import org.broadinstitute.dsde.workbench.leonardo.model.GoogleBucketUri.scheme
+import org.broadinstitute.dsde.workbench.leonardo.model.StringValueClass.LabelMap
+import spray.json.{DefaultJsonProtocol, DeserializationException, JsString, JsValue, JsonFormat, SerializationException}
 
 import scala.language.implicitConversions
 
-// maybe we want to get fancy later
-object ModelTypes {
-  type GoogleProject = String
-  type GoogleServiceAccount = String
-  type GoogleBucket = String
-  type GoogleBucketUri = String
-}
-
-object GoogleBucketUri {
-  def apply (bucketName: String, fileName: String): GoogleBucketUri = {
-    s"gs://${bucketName}/${fileName}"
-  }
+// this needs to be a Universal Trait to enable mixin with Value Classes
+// it only serves as a marker for StringValueClassFormat
+sealed trait StringValueClass extends Any
+case class ClusterName(string: String) extends AnyVal with StringValueClass
+case class GoogleProject(string: String) extends AnyVal with StringValueClass
+case class GoogleServiceAccount(string: String) extends AnyVal with StringValueClass
+case class GoogleBucket(string: String) extends AnyVal with StringValueClass
+case class OperationName(string: String) extends AnyVal with StringValueClass
+case class IP(string: String) extends AnyVal with StringValueClass
+case class BucketObject(string: String) extends AnyVal with StringValueClass
+case class FirewallRuleName(string: String) extends AnyVal with StringValueClass
+case class InstanceName(string: String) extends AnyVal with StringValueClass
+case class Zone(string: String) extends AnyVal with StringValueClass
+
+object StringValueClass {
+  type LabelMap = Map[String, String]
```

Comment:
not sure where this belongs, or if it's even useful

Reply:
I like this. String is such a basic type that "a map of them to more of them" is hard to contextualise compared to "it's the labels, dummy".

Reply:
👍